### PR TITLE
Look for options at the last position in MBQL clauses

### LINKED ISSUE #30280: Adding a row limit when there is a filter with a contains clause causes the filter to disappear
**Describe the bug**
When adding a Row Limit when there is a filter with a contains clause, the filter will disappear.
Additionally, if you remove the Row Limit, the filter will also disappear

**To Reproduce**
Steps to reproduce the behavior (if you can reproduce the bug using the Sample Database, we will find the issue faster):
1.  Start by Browsing Data, Sample Database
2. Choose Analytics Events table.
3. In the Notebook Editor, create a filter with a "contains" or "does not contain" option. I chose  Page URL does not contain "invite"
image
4.  Add a Row Limit.  I did a Row Limit of 3.
5.  Notice how the entire filters section disappears.
image

**Expected behavior**
Filters should not disappear

**Information about your Metabase Installation:**

Seeing this on master Built on 2023-04-20
Does not happen in 45.3 or 46.1

**Severity**
It's pretty annoying  I had to add back filters.

**Additional context**

### PR DESCRIPTION
**Pull request by metamben**
Fixes #30280.

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/30326)

### CODE REVIEW

File: src/metabase/lib/convert.cljc [hunk 1]
```diff
@@ -70,8 +70,14 @@
...
-  (let [[clause-type options & args] (lib.options/ensure-uuid mbql-clause)]
-    (into [clause-type options] (map ->pMBQL) args)))
+  (let [last-elem (peek mbql-clause)
+        last-elem-option? (map? last-elem)
+        [clause-type & args] (cond-> mbql-clause
+                               last-elem-option? pop)
+        options (if last-elem-option?
+                  last-elem
+                  {})]
+    (lib.options/ensure-uuid (into [clause-type] (map ->pMBQL) (cons options args)))))
```
Comment (snoe) [inline src/metabase/lib/convert.cljc:80]: I'm pretty sure `options` should not go through `->pMBQL`. The `update-vals` in `:dispatch-type/map` could easily make vector options be processed as an mbql clause.

**Review by snoe — CHANGES_REQUESTED** [review]

File: src/metabase/lib/convert.cljc (hunk 1 above)
Comment (metamben) [reply src/metabase/lib/convert.cljc:80]: > The `update-vals` in `:dispatch-type/map` could easily make vector options be processed as an mbql clause.

Do you know which kinds of maps that branch is expected to process? (TBH, I'm a bit uneasy with our practice of defining functions that can take anything and convert to something...)

Comment (snoe) [reply src/metabase/lib/convert.cljc:80]: @metamben  From what I've seen, I think it's largely about legacy MBQL and idempotency, but I can't speak to the intent.

Comment (metamben) [reply src/metabase/lib/convert.cljc]: @snoe I went with your proposal. After all, the original didn't convert option values either.

**Review by snoe — APPROVED** [review]

**Pull request merged**